Tactical/Dialogue Control.cpp

```diff
@@ -1973,99 +1914,85 @@ BOOLEAN SnitchCharacterDialogue( UINT8 ubCharacterNum, UINT16 usQuoteNum, INT32
 
 	return( TRUE );
 }
 
 
 BOOLEAN SpecialCharacterDialogueEvent( UINT32 uiSpecialEventFlag, UINT32 uiSpecialEventData1, UINT32 uiSpecialEventData2, UINT32 uiSpecialEventData3, INT32 iFaceIndex, UINT8 bUIHandlerID )
 {
-	DIALOGUE_Q_STRUCT				*QItem;
-
-	// Allocate new item
-	QItem = (DIALOGUE_Q_STRUCT *) MemAlloc( sizeof( DIALOGUE_Q_STRUCT ) );
-	memset( QItem, 0, sizeof( DIALOGUE_Q_STRUCT ) );
-
-	QItem->uiSpecialEventFlag		= uiSpecialEventFlag;
-	QItem->uiSpecialEventData		= uiSpecialEventData1;
-	QItem->uiSpecialEventData2	= uiSpecialEventData2;
-	QItem->uiSpecialEventData3	= uiSpecialEventData3;
-	QItem->iFaceIndex			= iFaceIndex;
-	QItem->bUIHandlerID		= bUIHandlerID;
-	QItem->iTimeStamp			= GetJA2Clock( );
+	DIALOGUE_Q_STRUCT QItem;
+	memset(&QItem, 0, sizeof(DIALOGUE_Q_STRUCT));
+	QItem.uiSpecialEventFlag	= uiSpecialEventFlag;
+	QItem.uiSpecialEventData	= uiSpecialEventData1;
+	QItem.uiSpecialEventData2	= uiSpecialEventData2;
+	QItem.uiSpecialEventData3	= uiSpecialEventData3;
+	QItem.iFaceIndex			= iFaceIndex;
+	QItem.bUIHandlerID			= bUIHandlerID;
+	QItem.iTimeStamp			= GetJA2Clock( );
 
 	// if paused state not already locked
 	if( gfLockPauseState == FALSE )
 	{
-		QItem->fPauseTime	 = fPausedTimeDuringQuote;
+		QItem.fPauseTime	 = fPausedTimeDuringQuote;
 	}
 
 	fPausedTimeDuringQuote = FALSE;
 
 	// Add to queue
-	ghDialogueQ = AddtoQueue( ghDialogueQ, &QItem );
-
+	ghDialogueQ.push(QItem);
 
 	return( TRUE );
 }
 
 BOOLEAN SpecialCharacterDialogueEventWithExtraParam( UINT32 uiSpecialEventFlag, UINT32 uiSpecialEventData1, UINT32 uiSpecialEventData2, UINT32 uiSpecialEventData3, UINT32 uiSpecialEventData4, INT32 iFaceIndex, UINT8 bUIHandlerID )
 {
-	DIALOGUE_Q_STRUCT				*QItem;
-
-	// Allocate new item
-	QItem = (DIALOGUE_Q_STRUCT *) MemAlloc( sizeof( DIALOGUE_Q_STRUCT ) );
-	memset( QItem, 0, sizeof( DIALOGUE_Q_STRUCT ) );
-
-	QItem->uiSpecialEventFlag		= uiSpecialEventFlag;
-	QItem->uiSpecialEventData		= uiSpecialEventData1;
-	QItem->uiSpecialEventData2	= uiSpecialEventData2;
-	QItem->uiSpecialEventData3	= uiSpecialEventData3;
-	QItem->uiSpecialEventData4	= uiSpecialEventData4;
-	QItem->iFaceIndex			= iFaceIndex;
-	QItem->bUIHandlerID		= bUIHandlerID;
-	QItem->iTimeStamp			= GetJA2Clock( );
+	DIALOGUE_Q_STRUCT QItem;
+	memset(&QItem, 0, sizeof(DIALOGUE_Q_STRUCT));
+	QItem.uiSpecialEventFlag	= uiSpecialEventFlag;
+	QItem.uiSpecialEventData	= uiSpecialEventData1;
+	QItem.uiSpecialEventData2	= uiSpecialEventData2;
+	QItem.uiSpecialEventData3	= uiSpecialEventData3;
+	QItem.uiSpecialEventData4	= uiSpecialEventData4;
+	QItem.iFaceIndex			= iFaceIndex;
+	QItem.bUIHandlerID			= bUIHandlerID;
+	QItem.iTimeStamp			= GetJA2Clock( );
 
 	// if paused state not already locked
 	if( gfLockPauseState == FALSE )
 	{
-		QItem->fPauseTime	 = fPausedTimeDuringQuote;
+		QItem.fPauseTime	 = fPausedTimeDuringQuote;
 	}
 
 	fPausedTimeDuringQuote = FALSE;
 
 	// Add to queue
-	ghDialogueQ = AddtoQueue( ghDialogueQ, &QItem );
-
+	ghDialogueQ.push(QItem);

```

Comment:
this is making a copy of QItem. 99% not likely to be a problem, unless this is hot code, which I doubt.

Reply:
This is about as far from hot code as Marselus Wallace is from okay